### DISCUSSION TITLE
Change of folder rights

### DISCUSSION THREAD

**Discussion by Serph91P** [MrBrax/LiveStreamDVR#150] [Ideas]
https://github.com/MrBrax/TwitchAutomator/blob/2d94852e32fa939c3eb4ff84ca50b7e75188d725/Dockerfile#L33

Could you change this a bit so this here works, I hope this would solve all problems with Docker:
Example:
```
FROM ubuntu

ARG USER_ID
ARG GROUP_ID

RUN addgroup --gid $GROUP_ID user
RUN adduser --disabled-password --gecos '' --uid $USER_ID --gid $GROUP_ID user
USER user
```

Because then we could use somthing like this when building the docker:
  --build-arg **USER_ID**=$(id -u) \
  --build-arg **GROUP_ID**=$(id -g) .

To get the correct user for everything.

**No need to use “chown”, and no annoying permission errors anymore!**

Here is the link where I got this from: [here](https://vsupalov.com/docker-shared-permissions/)

**Comment by MrBrax** [comment]
what about the built in node user that the node image creates?

**Reply by Serph91P** [reply]
The build in node user kinda blocks writing to files externaly. Like if I want to change a json because TA crashed or so. I have to connect to the docker and change it from there and cannot user for exanple windows go to this location and change the json because it says access denied. Also cannot create or copy files to it.

**Reply by MrBrax** [reply]
externally the data folder should be owned by 1000:1000, if you have an user/group with those id's, it should be able to write i think

**Comment by MrBrax** [comment]
also this would require cloning it yourself and building it for your platform, right? going via dockerhub wouldn't work

**Reply by Serph91P** [reply]
If you change it to that user ID with the variable It should still work or? Did I misread that information. Also no docker pro but this would generally fix the premission issues occuring. As far as I understand it all the other docker for Unraid are also hosted on Dockerhub. But have somekind of user thingy going on so files can be written in the mounted folders.

**Reply by MrBrax** [reply]
when fetching from dockerhub you don't build anything afaik, you download the images the bot built, and as such you can't modify the build process yourself with args

**Reply by Serph91P** [reply]
You shure about that? Because then Iam asking myself how would these changes then be implemented, because I can change these and see the difference. For example a port or smth.
![image](https://user-images.githubusercontent.com/8904984/165921910-cedb7440-0d45-44f1-b62e-5a9b3f93dde5.png)
This is from a Downloader on Unraid.

**Reply by MrBrax** [reply]
environment variables and mounts can be set, but you're never building anything. the build happens on github servers way before you download it, by which point the only thing that gets executed on your end when starting it is "yarn run start"

**Reply by MrBrax** [reply]
umask/puid/pgid must be something unraid specific, because i can't remember putting in such a thing anywhere

**Reply by Serph91P** [reply]
hmmm its linux specific. Will have another look at it.

**Comment by MrBrax** [comment]
also, for the created permissions, i think i read that the host os always handles the permissions, so when you mount a folder to the docker container, it retains the same permissions within. i no longer run chown/chmod on the data directory, so it should keep the same ones throughout.
